base_layer/core/src/proof_of_work/monero_rx/helpers.rs

```diff
@@ -59,7 +59,7 @@ pub fn monero_difficulty(header: &BlockHeader, randomx_factory: &RandomXFactory)
 fn get_random_x_difficulty(input: &[u8], vm: &RandomXVMInstance) -> Result<(Difficulty, Vec<u8>), MergeMineError> {
     let hash = vm.calculate_hash(input)?;
     debug!(target: LOG_TARGET, "RandomX Hash: {:?}", hash);
-    let difficulty = little_endian_difficulty(&hash);
+    let difficulty = little_endian_difficulty(&hash).map_err(|e| MergeMineError::DeserializeError(e.to_string()))?;
```

Comment:
This is the wrong error. 
I would just add the difficulty error to MergeMineError
and then just use`?`